core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -147,21 +148,23 @@ class PartKeyLuceneIndex(dataset: Dataset,
     logger.info(s"Started flush thread for lucene index on dataset=${dataset.ref} shard=$shardNum")
   }
 
-  def removeEntries(prunedPartitions: EWAHCompressedBitmap): Unit = {
-    val deleteQuery = IntPoint.newSetQuery(PartKeyLuceneIndex.PART_ID, prunedPartitions.toList)
-    indexWriter.deleteDocuments(deleteQuery)
-  }
-
   /**
     * Use to delete partitions that were ingesting before retention period
     * @return partIds of deleted partitions
     */
-  def removePartKeysEndedBefore(endedBefore: Long): IntIterator = {
+  def partIdsEndedBefore(endedBefore: Long): EWAHCompressedBitmap = {
     val collector = new PartIdCollector()
     val deleteQuery = LongPoint.newRangeQuery(PartKeyLuceneIndex.END_TIME, 0, endedBefore)
     searcherManager.acquire().search(deleteQuery, collector)
-    indexWriter.deleteDocuments(deleteQuery)
-    collector.intIterator()
+    collector.result
+  }
+
+  def removePartKeys(partIds: Array[Int]): Unit = {
```

Comment:
Consider passing in another EWAHCompressedBitmap instead of Array[Int], may save space.  I know this is probably for filtering out partitions.  This could be easily done with bitmap math.

Reply:
Will do.